### PR TITLE
Add grpc

### PR DESCRIPTION
**Pull request by ayaz-ashrapov**

### CODE REVIEW

File: raft.proto [hunk 1]
```diff
@@ -0,0 +1,52 @@
...
+    uint64 term = 1;
+	Command command = 2;
+}
+
+message Command {
+    int32 value = 1;
+}
+
+message AppendEntryResponse{
+    int32 receiver_id = 1;
```
Comment (ferzisdis) [inline raft.proto:32]: receiver_id should be removed in dto, because it is helper field for internal usage

File: raft.proto [hunk 2]
```diff
@@ -0,0 +1,52 @@
+syntax = "proto3";
```
Comment (ferzisdis) [inline raft.proto:1]: better to format all files with gofmt
there are some paddings issues

File: go.mod [hunk 3]
```diff
@@ -2,7 +2,15 @@ module github.com/Boar-D-White-Foundation/kvgo
 
 go 1.21
 
-require github.com/gin-gonic/gin v1.9.1
+require (
+	github.com/gin-gonic/gin v1.9.1
+	github.com/matryer/is v1.4.1
```
Comment (ferzisdis) [inline go.mod:7]: you added a framework for tests but not a single test :D 
suggest to remove "gin" and "is" frameworks for now

File: go.mod [hunk 4]
```diff
@@ -23,10 +31,11 @@ require (
...
-	golang.org/x/crypto v0.9.0 // indirect
-	golang.org/x/net v0.10.0 // indirect
-	golang.org/x/sys v0.8.0 // indirect
-	golang.org/x/text v0.9.0 // indirect
-	google.golang.org/protobuf v1.30.0 // indirect
+	golang.org/x/crypto v0.14.0 // indirect
+	golang.org/x/net v0.16.0 // indirect
+	golang.org/x/sys v0.13.0 // indirect
+	golang.org/x/text v0.13.0 // indirect
+	google.golang.org/grpc v1.60.0
```
Comment (ferzisdis) [inline go.mod:38]: it would be better to lift direct dependency up

File: cmd/kv-go/server_grpc.go [hunk 5]
```diff
@@ -0,0 +1,165 @@
...
+
+// HandleVoteRequest implements kvgo.RequestHandler.
+func (r *rafterServer) HandleVoteRequest(request kvgo.VoteRequest) {
+	go func() {
+		req := pb.VoteRequest{
+			CandidateId:   int32(request.CandidateId),
+			RequestInTerm: request.Term,
+		}
+
+		conn, err := grpc.Dial(r.cfg.Cluster[request.ReceiverId], grpc.WithTransportCredentials(insecure.NewCredentials()))
```
Comment (ferzisdis) [inline cmd/kv-go/server_grpc.go:63]: creating connection every time lead to problem with performance
suggest to create issue in GitHub for fix it later

File: cmd/kv-go/server_grpc.go [hunk 6]
```diff
@@ -0,0 +1,165 @@
...
+			return
+		}
+
+		rc := pb.NewRafterClient(conn)
+
+		//recipient := r.clients[request.ReceiverId]
+		resp, err := rc.AppendEntry(context.Background(), &req)
+
+		if err != nil {
+			slog.Error("can't append entry: %s", err)
```
Comment (ferzisdis) [inline cmd/kv-go/server_grpc.go:41]: slog is structured log
you need to change parameters format for: slog.Error("can't append entry", "error", err)
check doc https://pkg.go.dev/golang.org/x/exp/slog

File: cmd/kv-go/main.go [hunk 7]
```diff
@@ -3,28 +3,38 @@ package main
...
 	if err != nil {
-		slog.Error("can't parse config", "error", err)
+		slog.Error("can't parse config", err)
 		return
 	}
 
 	raft := kvgo.MakeRaft(*config, rtimer.Default())
-	rest := makeRestServer(config, &raft)
-	raft.SetHandler(&rest)
+	//rest := makeRestServer(config, &raft)
```
Comment (ferzisdis) [inline cmd/kv-go/main.go:32]: you can safely delete all rest implementation include files

**Review by ayaz-ashrapov — COMMENTED** [review]
Let"s ship it!

**Review by ferzisdis — COMMENTED** [review]
finished

**Review by ferzisdis — APPROVED** [review]

**Pull request merged**